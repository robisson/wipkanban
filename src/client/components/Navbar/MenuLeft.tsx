import * as React from "react";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import Button from "@material-ui/core/Button";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import { withStyles, StyleRulesCallback } from "@material-ui/core/styles";

const styles: StyleRulesCallback = () => ({
  navLeft: {
    marginLeft: 20
  }
});

type Props = {
  classes: any;
};

const MenuLeft: React.StatelessComponent<any> = ({ classes }: Props) => {
  return (
    <div className={classes.navLeft}>
      <Button
        color="inherit"
        size="small"
        style={{
          position: "relative",
          paddingTop: 15
        }}
      >
        <DeveloperBoardIcon />
        <small
          style={{
            position: "absolute",
            top: 0,
            left: 12,
            fontSize: 8
          }}
        >
          {"Guarida/Tecnologia da informação"}
        </small>
        Minhas atividades
        <ArrowDropDownIcon />
      </Button>
    </div>
  );
};

export default withStyles(styles)(MenuLeft);
