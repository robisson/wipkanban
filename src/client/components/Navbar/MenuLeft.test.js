import React from "react";
import MenuLeft from "./MenuLeft";
import { createShallow } from "@material-ui/core/test-utils";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import Button from "@material-ui/core/Button";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";

// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("MenuLeft Component", () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it("should render a div", () => {
    let wrapper = shallow(<MenuLeft />);
    expect(wrapper.type()).toEqual("div");
  });

  it("should render a Button containing two icons", () => {
    let wrapper = shallow(<MenuLeft />);
    expect(wrapper.find(Button).type()).toEqual(Button);
    expect(wrapper.find(DeveloperBoardIcon).type()).toEqual(DeveloperBoardIcon);
    expect(wrapper.find(ArrowDropDownIcon).type()).toEqual(ArrowDropDownIcon);
  });
});
