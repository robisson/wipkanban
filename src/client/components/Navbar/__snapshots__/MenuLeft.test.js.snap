// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLeft Component should render a Button containing two icons 1`] = `
<div
  className="MenuLeft-navLeft-1"
>
  <WithStyles(Button)
    color="inherit"
    size="small"
    style={
      Object {
        "paddingTop": 15,
        "position": "relative",
      }
    }
  >
    <pure(DeveloperBoardIcon) />
    <small
      style={
        Object {
          "fontSize": 8,
          "left": 12,
          "position": "absolute",
          "top": 0,
        }
      }
    >
      Guarida/Tecnologia da informação
    </small>
    Minhas atividades
    <pure(ArrowDropDownIcon) />
  </WithStyles(Button)>
</div>
`;

exports[`MenuLeft Component should render a div 1`] = `
<div
  className="MenuLeft-navLeft-1"
>
  <WithStyles(Button)
    color="inherit"
    size="small"
    style={
      Object {
        "paddingTop": 15,
        "position": "relative",
      }
    }
  >
    <pure(DeveloperBoardIcon) />
    <small
      style={
        Object {
          "fontSize": 8,
          "left": 12,
          "position": "absolute",
          "top": 0,
        }
      }
    >
      Guarida/Tecnologia da informação
    </small>
    Minhas atividades
    <pure(ArrowDropDownIcon) />
  </WithStyles(Button)>
</div>
`;
